### ISSUE TITLE
Questions about details of Multi-Class classification and monotonicity constraints 

### ISSUE THREAD

**Issue by SrayUM** [dmlc/xgboost#9647]
I have questions regarding how multi-class classification is optimized by the package and the monotonicity constraints:

(1) To optimize the multiple class (m classes), the Gradients and Hessians are calculated for each class and used to split the trees. So for each tree, the split is independent? If I want to use m binary models to replicate the multi-class one, I only need to use a customized objective function to calculate the softmax G and H and sent back to each model, is this correct?

(2) If I want to have different monotonicity constraints for each sub-model, how can I do that? Is the current package able to accommodate this demand?

Thank you a lot!
